ice-adapter/src/main/java/com/faforever/iceadapter/util/ExecutorHolder.java

```diff
@@ -0,0 +1,13 @@
+package com.faforever.iceadapter.util;
+
+import lombok.experimental.UtilityClass;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+@UtilityClass
+public class ExecutorHolder {
+    public ExecutorService getExecutor() {
+        return Executors.newVirtualThreadPerTaskExecutor();
```

Comment:
This creates a new executor every time. I guess we should keep one instance and return that all the time.

Reply:
`private final ExecutorService executor = ExecutorHolder.getExecutor();`
We create it once in the main class, and then distribute it to all other classes via IceAdapter.getExecutor()

Reply:
Ok, I guess that's not what Sheikah had in mind, but it's fine for now.